README.md

```diff
@@ -1,45 +1,91 @@
+`setInterval`: - more on setInterval,
+`clearInterval`: - more on clearInterval,
+`msg`: - Incoming message containing the payload from the previous step,
+`exports`: {},
+`messages`: - Utility for convenient message creation,
+`request`: - Http Client,
+`wait`: - wait,
+`emitter`: user to emit messages and errors
+
+## Code component usage Examples
+
+Use code is very simple, just do following:
+
+```JavaScript
+async function run(msg) {
+  console.log('Incoming message is %s', JSON.stringify(msg));
+  const body = { result : 'Hello world!' };
+  // You can emit as many data messages as required
+  await this.emit('data', { body });
+  console.log('Execution finished');
+}
+```
 
-> If you fail to upload you SSH Key you will get **permission denied** error during the deployment.
+Please note if you have a simple one-in-one-out function you can simply return a JSON object as a result
+of your function, it will be automatically emitted as data.
 
-## Getting Started
+## Common usage scenarios
 
-After registration and uploading of your SSH Key you can proceed to deploy it into our system. At this stage we suggest you to:
-* [Create a team](http://go2.elastic.io/manage-teams) to work on your new component. This is not required but will be automatically created using random naming by our system so we suggest you name your team accordingly.
-* [Create a repository](http://go2.elastic.io/manage-repositories) where your new component is going to *reside* inside the team that you have just created. For a simplicity you can name your repository **code-component** or **code*.
+### Doing complex data transformation
 
-```bash
-$ git clone https://github.com/elasticio/code-component.git code-component
+[JSONata](http://jsonata.org/) is great however sometimes it's easier to do things in JavaScript, if you want to transorm
+an incoming message with code, just use following sample:
 
-$ cd code-component
+```JavaScript
+async function run(msg) => {
+  addition : "You can use code",
+  keys : Object.keys(msg)   
+}
 ```
-Now you can edit your version of **code-component** component and change according to your needs - that is if you know what you are doing. Or you can just ``PUSH``it into our system to see the process in action:
-
-```bash
-$ git remote add elasticio your-created-team-name@git.elastic.io:code-component.git
 
-$ git push elasticio master
+### Calling an external REST API
+
+It's very simple to code a small REST API call out of the Code component, see following example:
+
+```JavaScript
+async function run(msg) {
+  const res = await request.get({
+    uri: 'https://api.elastic.io/v1/users',
+    auth: {
+      user: process.env.ELASTICIO_API_USERNAME,
+      pass: process.env.ELASTICIO_API_KEY
+    },
+    json: true  
+  });
+  return {
+    fullName: res.body.first_name + " " + res.body.last_name,
+    email: res.body.email,
+    userID: res.body.id    
+  }
+}
```

Comment:
Can you add some remarks that currently this component does not support snapshots or any form of credentials?